File: android/app/src/main/kotlin/com/example/hotel_reception/procesor/dto/update/HotelUpdate.kt
package com.example.hotel_reception.procesor.dto.update

import com.example.hotel_reception.procesor.dto.WorldSpeed
import java.util.*

data class HotelUpdate(
    val balance: Int,
    val monthIncome: Int,
    val date: Long,
    val rooms: List<RoomUpdate>,
    val worldSpeed: WorldSpeed,
    val dayIncome: Int,
)

File: lib/model/dto/hotel_update/hotel_update.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:hotel_reception/model/dto/hotel_update/room_update.dart';
import 'package:hotel_reception/model/dto/hotel_update/world_speed_dto.dart';
import 'package:hotel_reception/model/ui/world_speed.dart';

part 'hotel_update.freezed.dart';
part 'hotel_update.g.dart';

@freezed
class MonitorUpdate with _$MonitorUpdate {
  factory MonitorUpdate({
    required int balance,
    required int dayIncome,
    required int date,
    required List<RoomUpdate> rooms,
    required WorldSpeed worldSpeed, 
  }) = _MonitorUpdate;

  factory MonitorUpdate.fromJson(Map<String, dynamic> json) =>
      _$MonitorUpdateFromJson(json);
}
